types/set/array.pp

```diff
@@ -1,4 +1,4 @@
 #
 # @summary type to allow an array of ip addresses
 #
-type IPSet::Set::Array = Array[String]
+type IPSet::Set::Array = Variant[Array[String], Array[Integer]]
```

Comment:
The integers are ports? Then we could use Stdlib::Port> https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/types/port.pp

Reply:
The `IPSet::Set::Array` is has been used to define IP addresses, but now it can contain:
- IP address (Stdlib::IP::Address)
- port (Stdlib::Port)
- udp:port
- tcp:port

Making it more strict on the data type would involve a partial rewrite and most likely breaking changes.

Reply:
Or create each type as resource with strict data type would also be possible ?

Reply:
I think `type IPSet::Set::Array = Variant[Array[String], Array[Stdlib::Port]]` is fine for now. What do you think?